common/common-client/src/main/java/org/ow2/proactive/authentication/IAMLoginModule.java

```diff
@@ -0,0 +1,393 @@
+ */
+package org.ow2.proactive.authentication;
+
+import java.beans.BeanInfo;
+import java.beans.IntrospectionException;
+import java.beans.Introspector;
+import java.beans.PropertyDescriptor;
+import java.io.IOException;
+import java.util.Map;
+
+import javax.security.auth.Subject;
+import javax.security.auth.callback.Callback;
+import javax.security.auth.callback.CallbackHandler;
+import javax.security.auth.callback.UnsupportedCallbackException;
+import javax.security.auth.login.FailedLoginException;
+import javax.security.auth.login.LoginException;
+import javax.security.auth.spi.LoginModule;
+
+import org.apache.log4j.Logger;
+import org.jasig.cas.client.util.CommonUtils;
+import org.jasig.cas.client.util.ReflectUtils;
+import org.jasig.cas.client.validation.Assertion;
+import org.jasig.cas.client.validation.TicketValidationException;
+import org.jasig.cas.client.validation.TicketValidator;
+import org.ow2.proactive.authentication.iam.IAMRestClient;
+import org.ow2.proactive.authentication.principals.GroupNamePrincipal;
+import org.ow2.proactive.authentication.principals.UserNamePrincipal;
+
+
+/**
+ * Authentication based on ProActive IAM (Identity and Access Management) micro-service.
+ *
+ *
+ * based on Apereo CAS framework
+ *
+ *
+ * @author The ActiveEon Team
+ * @since ProActive Scheduling 8.2
+ */
+public class IAMLoginModule implements LoginModule {
+
+    /** Logger instance */
+    private static final Logger LOG = Logger.getLogger(IAMLoginModule.class.getName());
+
+    private static final String IAM_URL_KEY = "iamServerUrlPrefix";
+
+    /** JAAS authentication subject */
+    private Subject subject;
+
+    /** JAAS callback handler */
+    private CallbackHandler callbackHandler;
+
+    /** IAM Server URL */
+    private String iamServerUrlPrefix;
+
+    /** IAM REST request for tickets */
+    private String iamTicketRequest;
+
+    /** IAM Ticket/Token validator */
+    private String ticketValidatorClass;
+
+    /** IAM Ticket/Token validator extracted from ticketValidatorClass*/
+    private TicketValidator ticketValidator;
+
+    /** IAM client service */
+    private String service;
+
+    /** IAM assertion */
+    private Assertion assertion;
+
+    /** Name of the attribute in the CAS assertion that should be used for the username */
+    private String userAttributeName;
+
+    /** character used to separate many roles assigned to the user */
+    private String roleSeparator;
+
+    /** Name of the attribute in the CAS assertion that should be used for user role data */
+    private String roleAttributeName;
+
+    /** IAM response header used to get the SSO ticket */
+    private String ssoTicketHeader;
+
+    /** Boolean mentioning whether the JWT is signed */
+    private boolean jsonWebTokenSigned;
+
+    /** Boolean mentioning whether the JWT encrypted */
+    private boolean jsonWebTokenEncrypted;
+
+    /** JWT signature key */
+    private String jsonWebTokenSignatureKey;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the "jsonWebTokenSignatureKey" field and declare it as a local variable in the relevant methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1450)